Tests/UnitTests/Core/Propagator/AtlasStepperTests.cpp

```diff
@@ -12,228 +12,324 @@
 #include "Acts/MagneticField/ConstantBField.hpp"
 #include "Acts/Propagator/AtlasStepper.hpp"
 #include "Acts/Tests/CommonHelpers/FloatComparisons.hpp"
-
-namespace tt = boost::test_tools;
+#include "Acts/Utilities/Units.hpp"
 
 namespace Acts {
 namespace Test {
 
+using namespace Acts::UnitLiterals;
 using Covariance = BoundSymMatrix;
 using Jacobian = BoundMatrix;
+using Stepper = AtlasStepper<ConstantBField>;
+
+/// Simplified propagator state.
+struct MockPropagatorState {
+  MockPropagatorState(Stepper::State stepperState)
+      : stepping(std::move(stepperState)) {}
 
-/// @brief Simplified propagator state
-struct PropState {
-  /// @brief Constructor
-  PropState(AtlasStepper<ConstantBField>::State sState) : stepping(sState) {}
-  /// State of the Atlas stepper
-  AtlasStepper<ConstantBField>::State stepping;
-  /// Propagator options which only carry the relevant components
+  /// Stepper state.
+  Stepper::State stepping;
+  /// Propagator options with only the relevant components.
   struct {
-    double mass = 42.;
-    double tolerance = 1e-4;
+    double mass = 1_GeV;
+    double tolerance = 10_um;
```

Comment:
Does that play nicely with any internal magic numbers if the definitions ever change?

Reply:
I do not quite understand what you mean? Magic numbers inside the `AtlasStepper`? I chose typical values for our use case, which should always give reasonable results regardless of internal magic numbers, right?

Reply:
Ok, now I understand what you mean. At this point, I assume that everything has been updated to use the correct unit definitions. If they do not, than that is a bug and should be fixed.

Reply:
Well that's kind of the point. They possibly **can't** be fixed, because there might not be any information on how the magic numbers come to be.